### PR TITLE
switch underlying matrix library from gomatrix to mautrix-go

### PR DESCRIPTION
**Pull request by metskem**
Since the gomatrix library is no longer supported, I think we should switch to mautrix-go.

### CODE REVIEW

**Comment by sebhoss** [comment]
Thanks a lot!

AFAIK the room joining part was added so that the initial setup in a complete new rooms works fine, e.g. in our case we have to invite the bot account to a room and the bot has to somehow accept that invitation before being able to send messages. Does this work in the new library without doing an explicit join?

**Comment by metskem** [comment]
Ah, I missed that. And no, I don't think mautrix-go will allow uninvited members to send messages either. So that would have to be added back in again. 
In our test cases we explicitly invited the bot account and also logged in with the bot account to accept the invite.
If you have time please add it, or I can do it somewhere during the weekend or Monday.

**Comment by sebhoss** [comment]
Ok ok - let see who is faster :smile:

File: internal/matrix/client.go [hunk 1]
```diff
@@ -7,24 +7,20 @@ package matrix
...
 
 var (
-	joinRoomSuccessTotal = promauto.NewCounterVec(prometheus.CounterOpts{
-		Name: "matrix_alertmanager_receiver_join_room_success_total",
-		Help: "The total number of successful join room operations",
-	}, []string{"room"})
-	joinRoomFailureTotal = promauto.NewCounterVec(prometheus.CounterOpts{
-		Name: "matrix_alertmanager_receiver_join_room_failure_total",
-		Help: "The total number of failed join room operations",
-	}, []string{"room"})
```
Comment (sebhoss) [inline internal/matrix/client.go:27]: Can we add these back or do they somehow break something somewhere?

**Review by sebhoss — CHANGES_REQUESTED** [review]

**Review by sebhoss — APPROVED** [review]

**Comment by sebhoss** [comment]
@metskem thanks a lot!

**Pull request merged**